include/pagmo/utils/hv_algos/hv_hvwfg.hpp

```diff
@@ -0,0 +1,413 @@
+ * Computation, IEEE Transactions on 16.1 (2012): 86-95."
+ * @see "Lyndon While and Lucas Bradstreet. Applying the WFG Algorithm To Calculate Incremental Hypervolumes. 2012 IEEE
+ * Congress on Evolutionary Computation. CEC 2012, pages 489-496. IEEE, June 2012."
+ */
+class hvwfg : public hv_algorithm
+{
+public:
+    /// Constructor
+    /**
+     * @param stop_dimension The stop dimension
+     */
+    hvwfg(unsigned int stop_dimension = 2u) : hv_algorithm(), m_current_slice(0), m_stop_dimension(stop_dimension)
+    {
+        if (stop_dimension < 2u) {
+            pagmo_throw(std::invalid_argument, "Stop dimension for WFG must be greater than or equal to 2");
+        }
+    }
+
+    /// Compute hypervolume
+    /**
+    * Computes the hypervolume using the WFG algorithm.
+    *
+    * @param points vector of points containing the D-dimensional points for which we compute the hypervolume
+    * @param r_point reference point for the points
+    *
+    * @return hypervolume.
+    */
+    double compute(std::vector<vector_double> &points, const vector_double &r_point) const
+    {
+        allocate_wfg_members(points, r_point);
+        double hv = compute_hv(1);
+        free_wfg_members();
+        return hv;
+    }
+
+    /// Contributions method
+    /**
+    * This method employs a slightly modified version of the original WFG algorithm to suit the computation of the
+    * exclusive contributions.
+    * It differs from the IWFG algorithm (referenced below), as we do not use the priority-queueing mechanism, but
+    * compute every exclusive contribution instead.
+    * This may suggest that the algorithm for the extreme contributor itself reduces to the 'naive' approach. It is not
+    * the case however,
+    * as we utilize the benefits of the 'limitset', before we begin the recursion.
+    * This simplifies the sub problems for each exclusive computation right away, which makes the whole algorithm much
+    * faster, and in many cases only slower than regular WFG algorithm by a constant factor.
+    *
+    * @see "Lyndon While and Lucas Bradstreet. Applying the WFG Algorithm To Calculate Incremental Hypervolumes. 2012
+    * IEEE Congress on Evolutionary Computation. CEC 2012, pages 489-496. IEEE, June 2012."
+    *
+    * @param points vector of points containing the D-dimensional points for which we compute the hypervolume
+    * @param r_point reference point for the points
+    *
+    * @return the single contributions
+    */
+    std::vector<double> contributions(std::vector<vector_double> &points, const vector_double &r_point) const
+    {
+        std::vector<double> c;
+        c.reserve(points.size());
+
+        // Allocate the same members as for 'compute' method
+        allocate_wfg_members(points, r_point);
+
+        // Prepare the memory for first front
+        double **fr = new double *[m_max_points];
```

Comment:
You mean memory allocation failures?
I would as a waiver for this PR as Its full of these things and I am not sure it would make sense to go through the whole code and fix these things. Also, I would have to ask you how to fix them properly ....